UserCheckPassword/User.js

```diff
@@ -0,0 +1,11 @@
+function User(username, password) {
+    this.username = username;
+    this.password = password;
+}
+
+
+User.prototype.checkPassword = function checkPassword(password) {
+    return typeof password === 'string' && this.password === password ? true : false;
```

Comment:
`===` already checks to make sure the type of comparison is same. 
Also the results of `this.password === password` is already a boolean therefore does not need a ternary.